### PR TITLE
storage: include segment and table info in more logs

### PR DESCRIPTION
**Pull request by breezewish**
### What problem does this PR solve?

Issue Number: ref #5726

### What is changed and how it works?

Attach the table_id and segment_id in more storage modules for easier debugging.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below): WIP

A. All segment logs now contains table_id, in case of multiple tables contains same segment id:

image

The recommended grep command would be:

```
grep 'segment_id=X\>' | grep 'table_id=X\>'
```

B. DeltaValueSpace and StableValueSpace now contains segment_id and table_id. A useful scenario is that we can see flush operations when grepping by the segment_id:

image

C. SSTFilesToDTFilesOutputStream now contains table_id. When grepping by the table we can know more information:

image

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/StableValueSpace.h [hunk 1]
```diff
@@ -141,14 +157,14 @@ class StableValueSpace : public std::enable_shared_from_this<StableValueSpace>
 
         ColumnCachePtrs column_caches;
 
-        Snapshot()
-            : log(&Poco::Logger::get("StableValueSpace::Snapshot"))
+        Snapshot(StableValueSpacePtr stable_)
+            : stable(stable_)
+            , log(Logger::get("StableValueSpace::Snapshot", fmt::format("<{}>", stable->log_prefix)))
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/StableValueSpace.h:162]: Maybe use `stable->log->identifier()` instead of saving `log_prefix` in `StableValueSpace`?

Comment (breezewish) [reply dbms/src/Storages/DeltaMerge/StableValueSpace.h]: Updated. Now there are still some `log_prefix` variables remain, when we need to concat multiple properties together. We may resolve it in https://github.com/pingcap/tiflash/issues/6068

**Review by JaySon-Huang — APPROVED** [review]
LGTM

File: dbms/src/Storages/DeltaMerge/Delta/ColumnFilePersistedSet.h [hunk 2]
```diff
@@ -69,19 +69,19 @@ class ColumnFilePersistedSet : public std::enable_shared_from_this<ColumnFilePer
...
+    LoggerPtr log;
 
 private:
     inline void updateColumnFileStats();
 
     void checkColumnFiles(const ColumnFilePersistedLevels & new_column_file_levels);
 
 public:
-    explicit ColumnFilePersistedSet(PageId metadata_id_, const ColumnFilePersisteds & persisted_column_files = {});
+    explicit ColumnFilePersistedSet(const std::string & log_prefix_, PageId metadata_id_, const ColumnFilePersisteds & persisted_column_files = {});
```
Comment (Lloyd-Pottiger) [inline dbms/src/Storages/DeltaMerge/Delta/ColumnFilePersistedSet.h:80]: use `String`?

**Review by Lloyd-Pottiger — APPROVED** [review]

File: dbms/src/Storages/DeltaMerge/Delta/ColumnFilePersistedSet.h (hunk 2 above)
Comment (breezewish) [reply dbms/src/Storages/DeltaMerge/Delta/ColumnFilePersistedSet.h:80]: Not sure about the idiom, but seems that in our code base both `std::string` and `String` are heavily used :/

**Pull request merged**